annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +306,241 @@ def test_validate_user_actions_invalid_states(
+    expected_result = "task1"
+
+    mock_query = MagicMock()
+    mock_filter = MagicMock()
+
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_filter
+    mock_filter.first.return_value = expected_result
+
+    result = read_annotation_task(mock_session, task_id=1, tenant="tenant_1")
+
+    assert result == expected_result
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_query.filter.assert_called_once()
+    mock_filter.first.assert_called_once()
+
+
+def test_filter_tasks_db_no_additional_filters(mock_session: Mock):
+    request = TaskFilter()
+    tenant = "test_tenant"
+    token = "test_token"
+
+    with patch(
+        "annotation.tasks.services.map_request_to_filter"
+    ) as mock_map_request_to_filter, patch(
+        "annotation.tasks.services.remove_additional_filters"
+    ) as mock_remove_additional_filters, patch(
+        "annotation.microservice_communication."
+        "assets_communication.get_files_by_request"
+    ) as mock_get_files_by_request, patch(
+        "annotation.jobs.services.get_jobs_by_name"
+    ) as mock_get_jobs_by_name, patch(
+        "annotation.tasks.services.form_query"
+    ) as mock_form_query, patch(
+        "annotation.tasks.services.paginate"
+    ) as mock_paginate:
+
+        mock_query = MagicMock(return_value=[])
+        mock_session.query.return_value = mock_query
+
+        mock_query.filter.return_value = mock_query
+        mock_paginate.return_value = ([], MagicMock())
+
+        mock_map_request_to_filter.return_value = {
+            "filters": [],
+            "sorting": [],
+        }
+        mock_remove_additional_filters.return_value = {}
+        mock_get_files_by_request.return_value = {}
+        mock_get_jobs_by_name.return_value = {}
+        mock_form_query.return_value = (MagicMock(), MagicMock())
+        mock_paginate.return_value = []
+
+        result = filter_tasks_db(mock_session, request, tenant, token)
+
+        assert result == ([], {}, {})
+
+
+def test_filter_tasks_db_file_and_job_name(mock_session: Mock):
+    additional_filters = {"file_name": ["file1"], "job_name": ["job1"]}
+    files_by_name = {"file1": 1}
+    jobs_by_name = {"job1": 1}
+    expected_result = ([MagicMock()], {"file1": 1}, {"job1": 1})
+    request = TaskFilter()
+    tenant = "test_tenant"
+    token = "test_token"
+    with patch(
+        "annotation.tasks.services.map_request_to_filter"
+    ) as mock_map_request_to_filter, patch(
+        "annotation.tasks.services.remove_additional_filters"
+    ) as mock_remove_additional_filters, patch(
+        "annotation.tasks.services.get_files_by_request"
+    ) as mock_get_files_by_request, patch(
+        "annotation.tasks.services.get_jobs_by_name"
+    ) as mock_get_jobs_by_name, patch(
+        "annotation.tasks.services.form_query"
+    ) as mock_form_query, patch(
+        "annotation.tasks.services.paginate"
+    ) as mock_paginate:
+
```

Comment:
remove this blank space, please, same applies for the rest below this comment. You can keep the ones after mocks